main.go

```diff
@@ -59,9 +60,12 @@ func main() {
 
 	flagSet := flag.NewFlagSet("cluster-machine-approver", flag.ExitOnError)
 
-	klog.InitFlags(flagSet)
+	klog.InitFlags(nil)
+	goflag.Parse()
+	flag.CommandLine.AddGoFlagSet(goflag.CommandLine)
+
 	flagSet.StringVar(&cliConfig, "config", "", "CLI config")
-	flagSet.StringVar(&APIGroup, "apigroup", "machine.openshift.io", "API group for machines, defaults to machine.openshift.io")
+	flagSet.StringSliceVar(&APIGroups, "apigroup", []string{"machine.openshift.io"}, "API group for machines, defaults to machine.openshift.io. This option can be given multiple times")
```

Comment:
How about having a new flag `api-group-version`, then deprecating this old flag? I think it would be better to be more explicit about what we are expecting here